### DISCUSSION TITLE
Switch on valve board

### DISCUSSION THREAD

**Discussion by cxrodgers** [wehr-lab/autopilot#88] [General]
In the old google group days, Jonny posted a file called valve_board.pdf (contained in poke_files_200912.zip) with a little schematic for a pushbutton switch to manually trigger the valve to open without using the raspberry pi (e.g., for flushing). Can't figure out how to attach the file, lmk if it's needed and I'll share it by dropbox

I'm starting to wonder if there's an error on this sheet. It calls the switch "TE 1825910-7". But when I looked this up, I got to this page:
https://www.digikey.com/en/products/detail/te-connectivity-alcoswitch-switches/1825910-7/1731414

First off, on the webpage it looks like a SPST switch (open or closed). In Jonny's schematic, it looks like SPDT (what we could call in plumbing a "tee"). Moreover, the current limit is a measly 50mA, but I think these LHDA valves go up to like 110mA or something, right? (Valve datasheet says max power consumption is 550 mW.) 

I'm wondering if the part number is incorrect on the schematic? Regardless, do people recommend running the valve current directly through the switch (as in the schematic), or doing something fancier where the switch just manipulates the base voltage on the transistor, but doesn't directly pass the valve current? Thanks for any tips or thoughts!!

**Comment by cxrodgers** [comment]
Oh never mind about the current limit being exceeded ... It's apparently 50 mA at 24VDC, which should be equivalent to plenty of current at 5VDC. Still wondering about the SPDT vs SPST thing though...

**Reply by cxrodgers** [reply]
Pretty proud of this PCB I laid out for the nosepoke. It was inspired by valve_board, but this one is for a single nosepoke, and also incorporates some other features I wanted. Anyone can purchase from OSH Park! But let me know before you do because I haven't tested it yet.
https://oshpark.com/shared_projects/vrsIkbtD

**Comment by sneakers-the-rat** [comment]
I'm sure what whatever you're seeing was the result of me not knowing best practices or how to draw the diragrams, so i'm glad you flagged this (I think aggregating this kind of technical knowledge is really important!)

I think you're right, my diagram is wrong (attached for reference) -- i was trying to communicate that when the button is unpressed/open, the circuit goes through the transistor, but when it's pressed/closed it is shorted to ground. It should probably be changed s.t., for each of the switches, the (1) point should just be connected to the (2) point and changed to a spst symbol.

I love the poke PCB!!!!!!!! I've started a (yet unadvertised) wiki for gathering technical knowledge, hardware designs, etc. here: https://wiki.auto-pi-lot.com/ . I've been bogged down by needing to do a bunch of degree requirements so i can finish this dang thing, and more recently by another wild-eyed infrastructure manifesto, so plz no judge its very infantile state and confusing organization. The plan is to fill this in as a semantically linked wiki, so, for example, someone could go looking for different nosepokes and be able to see all the different available designs, but one could also browse all the PCBs, parts submitted by specific users, designs & build guides that the parts are used in, etc. So, if you were to make a page with the `[[Part_Type::Nosepoke]]` it would show up here: https://wiki.auto-pi-lot.com/index.php/Nosepoke (which would in turn be part of a parent page that showed all the different `Part_Type`s . 

[valve_board.pdf](https://github.com/wehr-lab/autopilot/files/6261051/valve_board.pdf)
